### PR TITLE
add high level architecture doc

### LINKED ISSUE #1606: Brigade Design: A high-level explanation of how Brigade is designed.

### PR DESCRIPTION
**Pull request by krancour**
Fixes #1606

### CODE REVIEW

**Review by vdice — APPROVED** [review]
Do we want to place the image assets in docs/static/img as with the rest?  No strong opinion, just noticed that seemed to be the precedent.  I can also see it being nice to have them in-line with the doc, as is currently the case.

Otherwise, a few suggestions, also optional.

Well-written, as always!  LGTM

File: docs/content/topics/design.md [hunk 1]
```diff
@@ -8,244 +8,196 @@ aliases:
...
+how they integrate with one another to provide an event-driven scripting
+platform. It also provides some historical context for how maintainers arrived
+at the current architecture.
 
-# Brigade Design
+## Terminology
 
-_This is a living document, and is kept up to date with the current state of
-Brigade. It is a high-level explanation of the Brigade design._
+It is useful to begin with an implementation neutral description of Brigade's
```
[the remark this answers is not in the document]
Comment (krancour) [reply docs/content/topics/design.md:18]: I'm wishy washy on this one as well. I am fine erring on the side of removing it, so I will.

File: docs/content/topics/design.md [hunk 2]
```diff
@@ -8,244 +8,196 @@ aliases:
...
+## High Level Architecture
+
+The following depicts the major components of Brigade:
+
+<center>
+  <img src="architecture.png"/>
+</center>
+
+The following sections describe each of these components in greater detail. They
+are presented in an order that is intended accelerate understanding of the high
```
[the remark this answers is not in the document]
Comment (krancour) [reply docs/content/topics/design.md:59]: I am a grammar nerd and I'm embarrassed that I don't know the actual rule for this...

I guess for this and [this](https://github.com/brigadecore/brigade/pull/1710#discussion_r741447537) I'll research it a bit before finalizing.

File: docs/content/topics/design.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (krancour) [reply docs/content/topics/design.md:18]: After a review of these fairly concise rules, I think I will let this one stand.

https://www.grammarly.com/blog/hyphen/

File: docs/content/topics/design.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (krancour) [reply docs/content/topics/design.md:59]: Per this https://www.grammarly.com/blog/hyphen/ yes... this one should be changed.

File: docs/content/topics/design.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (krancour) [reply docs/content/topics/design.md:18]: D'oh. I read the suggested change backwards.

The hyphen _does_ belong here. "Let it stand" was in reference to _leaving_ it in place. But it isn't there...

I will fix.

[the remark this answers is not in the document]
Comment (krancour) [reply docs/content/topics/design.md]: Fixed for real now.

**Comment by krancour** [comment]
> Do we want to place the image assets in docs/static/img as with the rest? 

Good catch! Their current location was not meant to be final. This was an oversight.

Fixing.

**Review by vdice — APPROVED** [review]
LGTM!

**Pull request merged**